contributors/design-proposals/node/limit-node-object-self-control.md

```diff
@@ -0,0 +1,90 @@
+register a compromised node with label `foo/dedicated=customer-info-app`. The
+scheduler will then bind `customer-info-app` to the compromised node potentially
+giving the intruder easy access to the PII.
+
+This attack also extends to secrets. Suppose company `foo` runs their outward
+facing nginx on dedicated nodes to reduce exposure to the company's publicly
+trusted server certificates. They use the secret mechanism to distribute the
+serving certificate key. An intruder captures the dedicated nginx workload in
+the same way and can now use the node certificate to read the company's serving
+certificate key.
+
+### Gaining Access to Arbitrary Serving Certificates
+
+Suppose company `foo` uses TLS for server authentication between internal
+microservices. The company uses the Kubernetes certificates API to provision
+these workload certificates for workload `bar` and trust is rooted to the
+cluster's root certificate authority.
+
+When [kubelet server certificate
+rotation](https://github.com/kubernetes/features/issues/267) is complete, the
+same API will be used to provision serving certificates for kubelets. The design
+expects to cross-reference the addresses reported in the NodeStatus with the
+subject alternative names in the certificate signing request to validate the
+certificate signing request.
+
+An intruder can easily register a node with a NodeAddress `bar` and use this
+certificate to MITM all traffic to service `bar` the flows through kube-proxy on
+that node.
+
+## Proposed Solution
+
+In many environments, we can improve the situation by centralizing reporting of
+these node attributes to a more trusted source and disallowing reporting of
+these attributes from the kubelet.
+
+We can scope down the initial Node object creation by moving to a centralized
+controller model. In many deployment environments, the sensitive attributes of a
+Node object discussed above ("labels", "taints", "addresses") are discoverable
+by consulting a machine database (e.g. the GCE API). Using the
+[initializer](admission_control_extension.md) mechanism, a centralized
+controller can register an initializer for the node object and build the
+sensitive fields by consulting the machine database. The
+`cloud-controller-manager` is an obvious candidate to house such a controller.
+
+We can scope down subsequent updates of the Node object by moving control of
+sensitive fields into the NodeRestriction admission controller. For backwards
+compatibility we can begin by zero'ing updates to sensitive fields and after the
+kubelet compatibility window has expired, begin to return 403s. This also has
+the nice property of giving us fine grained control over field/values of
+updates. For example, in this model we could easily allow a kubelet to update
+it's `OutOfMemory` taint and disallow the kubelet from updating its `dedicated`
+taint.
+
+In this design we assume that the infrastructure has already validated caller ID
```

Comment:
Hi, how is it guaranteed that caller id is trusted. 
`nodeclient - a request by a user for a client certificate with O=system:nodes and CN=system:node:(node name)`
The `nodeclient` provides the CSR request and that request is coming from kubelet also. Who is creating this client cert? kubeadm?

Reply:
It's not guaranteed. I'm saying for the purposes of this design, we should assume that the node authentication credentials are trusted. Hardening that is a separate topic and out of scope of this proposal.